soc/xtensa/esp32s2/Kconfig.soc

```diff
@@ -69,6 +82,98 @@ config ESP32S2_DATA_CACHE_SIZE
 	default 0x4000 if ESP32S2_DATA_CACHE_16KB
 	default 0x0000
 
+config ESP_SPIRAM
+	bool "Support for external, SPI-connected RAM"
+	help
+	  This enables support for an external SPI RAM chip, connected in
+	  parallel with the main SPI flash chip.
+
+config ESP_HEAP_MIN_EXTRAM_THRESHOLD
+	int "Minimum threshold for external RAM allocation"
+	default 8192
+	range 1024 131072
+	depends on ESP_SPIRAM
+	help
+	  Threshold to decide if memory will be allocated from DRAM
+	  or SPIRAM. If value of allocation size is less than this value,
+	  memory will be allocated from internal RAM.
+
+menu "SPI RAM config"
+	depends on ESP_SPIRAM
+
+	choice SPIRAM_TYPE
+		prompt "Type of SPI RAM chip in use"
+		default SPIRAM_TYPE_ESPPSRAM16
+
+		config SPIRAM_TYPE_ESPPSRAM16
+			bool "ESP-PSRAM16 or APS1604"
+
+		config SPIRAM_TYPE_ESPPSRAM32
+			bool "ESP-PSRAM32 or IS25WP032"
+
+		config SPIRAM_TYPE_ESPPSRAM64
+			bool "ESP-PSRAM64 or LY68L6400"
+
+	endchoice # SPIRAM_TYPE
+
+	config ESP_SPIRAM_SIZE
+		int "Size of SPIRAM part"
+		default 2097152 if SPIRAM_TYPE_ESPPSRAM16
+		default 4194304 if SPIRAM_TYPE_ESPPSRAM32
+		default 8388608 if SPIRAM_TYPE_ESPPSRAM64
+		help
+		  Specify size of SPIRAM part.
+		  NOTE: If SPIRAM size is greater than 4MB, only
+		  lower 4MB can be allocated using k_malloc().
+
+	menu "PSRAM clock and cs IO for ESP32S2"
+		depends on ESP_SPIRAM
+
+		config DEFAULT_PSRAM_CLK_IO
+			int "PSRAM CLK IO number"
+			range 0 33
+			default 30
+			help
+			  The PSRAM CLOCK IO can be any unused GPIO, user can config
+			  it based on hardware design.
+
+		config DEFAULT_PSRAM_CS_IO
+			int "PSRAM CS IO number"
+			range 0 33
+			default 26
+			help
+			  The PSRAM CS IO can be any unused GPIO, user can config it
+			  based on hardware design.
+
+	endmenu
+
+	choice SPIRAM_SPEED
+		prompt "Set RAM clock speed"
+		default SPIRAM_SPEED_40M
+		help
+		  Select the speed for the SPI RAM chip.
+
+		config SPIRAM_SPEED_80M
+			bool "80MHz clock speed"
+
+		config SPIRAM_SPEED_40M
+			bool "40Mhz clock speed"
+
+		config SPIRAM_SPEED_26M
+			bool "26Mhz clock speed"
+
+		config SPIRAM_SPEED_20M
+			bool "20Mhz clock speed"
+
+	endchoice
+
+
+	config SPIRAM
+		bool
+		default y
```

Comment:
There is no indentation in Kconfig.

I'm surprised this passed CI.

Perhaps the code is not exercised by CI?

Reply:
Thanks for pointing this. Fixed.